cpp/src/table/table.cpp

```diff
@@ -33,10 +33,13 @@ table::table(table const& other) : _num_rows{other.num_rows()} {
 // Move the contents of a vector `unique_ptr<column>`
 table::table(std::vector<std::unique_ptr<column>>&& columns)
     : _columns{std::move(columns)} {
-  CUDF_EXPECTS(columns.size() > 0, "Invalid number of columns");
-  _num_rows = columns[0]->size();
-  for (auto const& c : _columns) {
-    CUDF_EXPECTS(c->size() == num_rows(), "Column size mismatch.");
+  if(_columns.size() != 0) {
+    _num_rows = _columns[0]->size();
+    for (auto const& c : _columns) {
```

Comment:
I realized that we should be doing extra safety checks and making sure that the contents of the unique_ptr are valid. 
```suggestion
    for (auto const& c : _columns) {
       CUDF_EXPECTS(c, "Unexpected null column");
```